File: workout-api/src/main/java/tom/projects/monkeylog/repository/workout/ProgramRepository.java
package tom.projects.monkeylog.repository.workout;

import org.springframework.data.jpa.repository.JpaRepository;
import tom.projects.monkeylog.model.workout.Program;

import java.util.List;

public interface ProgramRepository extends JpaRepository<Program, Long> {
    List<Program> findAllByUserId(Long userId);

}

File: backend/src/main/kotlin/tom/com/workout/logbook/repository/ProgramRepository.kt
package tom.com.workout.logbook.repository

import org.springframework.data.jpa.repository.JpaRepository
import tom.com.workout.logbook.model.workout.Program
import java.util.*

interface ProgramRepository : JpaRepository<Program, UUID> {
    fun findAllByUserId(userId: UUID): List<Program>
}
